contrib/signet/miner

```diff
@@ -93,10 +94,20 @@ def finish_block(block, signet_solution, grind_cmd):
         block.rehash()
     return block
 
-def generate_psbt(tmpl, reward_spk, *, blocktime=None, poolid=None):
-    signet_spk = tmpl["signet_challenge"]
+def generate_psbt(block, signet_spk):
```

Comment:
Better to move `def generate_psbt(..)` down to its code, rather than moving the code up?